File: Upcoder/Other/Tam giác/LTCB-GK1-S7-Cau1.cs
using System;

class Program
{
    static void Main()
    {
        while (true)
        {
            // Đọc vào 3 số nguyên a, b, c
            string input = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(input))
                break;

            string[] parts = input.Split(' ');
            if (parts.Length < 3)
                break;

            int a = int.Parse(parts[0]);
            int b = int.Parse(parts[1]);
            int c = int.Parse(parts[2]);

            // Kiểm tra tính hợp lệ của 3 cạnh
            if (a > 0 && b > 0 && c > 0 && (a + b > c) && (a + c > b) && (b + c > a))
            {
                int perimeter = a + b + c;
                Console.WriteLine(perimeter);
            }
            else
            {
                Console.WriteLine("NO");
            }
        }
    }
}

File: Upcoder/Other/Tam giác/LTCB-GK1-S7-Cau1.py
while True:
    # Nhập vào 3 số nguyên a, b, c
    input_expression = input()
    if not input_expression:
        break

    parts = input_expression.split()
    if len(parts) < 3:
        break

    a = int(parts[0])
    b = int(parts[1])
    c = int(parts[2])

    # Kiểm tra tính hợp lệ của 3 cạnh
    if a > 0 and b > 0 and c > 0 and (a + b > c) and (a + c > b) and (b + c > a):
        perimeter = a + b + c
        print(perimeter)
    else:
        print("NO")
